virtual-desktop/src/app/plugin-manager/plugin-factory/angular2/angular2-plugin-factory.ts

```diff
@@ -107,24 +107,30 @@ export class Angular2PluginFactory extends PluginFactory {
     const scriptUrl = Angular2PluginFactory.getAngularComponentsURL(pluginDefinition);
 
     return new Promise((resolve, reject) => {
-      (window as any).require([scriptUrl],
-        (components: MvdNativeAngularPluginComponentDefinition) => {
-          const factoryDefs = components.getComponentFactoryDefinitions(pluginDefinition);
-          factoryDefs.forEach((factory: AngularComponentFactoryDefinition) => {
-            const componentFactory = new SimpleAngularComponentFactory(this.compiler, this.applicationRef, this.injector, factory.componentScriptUrl, factory.componentClass, factory.capabilities);
-
-            console.log(`Registering component factory for plugin ${pluginDefinition.getIdentifier()}:`);
-            console.log(componentFactory);
-
-            ZoweZLUX.registry.registerComponentFactory(componentFactory);
-
+      if (pluginDefinition.hasComponents()) {
+        (window as any).require([scriptUrl],
+          (components: MvdNativeAngularPluginComponentDefinition) => {
+            const factoryDefs = components.getComponentFactoryDefinitions(pluginDefinition);
+            factoryDefs.forEach((factory: AngularComponentFactoryDefinition) => {
+              const componentFactory = new SimpleAngularComponentFactory(this.compiler, this.applicationRef, this.injector,
+              factory.componentScriptUrl, factory.componentClass, factory.capabilities);
+
+              console.log(`Registering component factory for plugin ${pluginDefinition.getIdentifier()}:`);
+              console.log(componentFactory);
+
+              ZoweZLUX.registry.registerComponentFactory(componentFactory);
+
+              resolve();
+            });
+          },
+          (failure: any) => {
+            console.log(`No component definition for plugin ${pluginDefinition.getIdentifier()}`);
             resolve();
           });
-        },
-        (failure: any) => {
-          console.log(`No component definition for plugin ${pluginDefinition.getIdentifier()}`);
+        } else {
+          console.log('This plugin does not use Zowe components');
```

Comment:
This needs to be more descriptive, otherwise you will get tons of "this plugin does not ..." and you'll be thinking "which one"?
Try:
`Plugin (${pluginDefinition.getIdentifier()}) didn't expose Zowe components. Component loading skipped`